tubesync/tubesync/gunicorn.py

```diff
@@ -22,7 +22,9 @@ def get_bind():
 
 
 workers = get_num_workers()
-timeout = 30
+timeout = 600
```

Comment:
This isn't really a sensible fix. The issue really is the source deletion needs to be moved to a background worker task and made async. Increasing the timeout here is a hack and 600 seconds still won't be long enough for anyone deleting a massive channel on slow hardware.

Note also there's a fixed timeout in the nginx config that pairs with this. Really, the web UI should be instant.

Obviously I can see that this might be "better" than the current status quo, but it's not really a fix.

Reply:
I don't disagree. I'll do a memdump to try and see exactly what is sitting in memory. If I can see anything obvious I'll test that out. I tried modifying background_tasks but the fix I applied to that caused errors in the log. I think a significant amount of the task data is sat in memory and never dropped.

I'm not familiar enough to replace the background_task component. If you do accept this workaround, it's important that this workaround is removed once the underlying issue is fixed.

You're also right that there is a limit to this workaround. I'm estimating this should support channels up to around 9000 videos on slower hardware.

Reply:
Moving to a background worker task isn't simple. I tried breaking the delete-source task up by creating a task per media item in the source, and just moving the delete-source to a background task. Either way, the gunicorn worker times out and the task is interrupted.

Reply:
Heh, yeah if this was just a simple task of wrapping some functions in `@background` decorators I'd have fixed this myself a long time ago...

Reply:
I was surprised when you suggested moving it to a background worker task but I tried it regardless. 

With regards to replacing the background queue, Is celery actually implemented? I can see s6-overlay is the entry point to the container and a celery supervisor there, but I can't see it in the pipfile so can't add a celery task decorator. I'm hoping a short term fix is to run both the background task and another DBMS backed task queue at the same time, with the option to slowly migrate tasks to it. I'm looking at Huey if celery isn't implemented.